engine/execution/state/delta/delta.go

```diff
@@ -10,46 +10,37 @@ import (
 
 // A Delta is a record of ledger mutations.
 type Delta struct {
-	Data map[string]flow.RegisterEntry
+	Data map[flow.RegisterID]flow.RegisterValue
 }
 
 // NewDelta returns an empty ledger delta.
 func NewDelta() Delta {
 	return Delta{
-		Data: make(map[string]flow.RegisterEntry),
+		Data: make(map[flow.RegisterID]flow.RegisterValue),
 	}
 }
 
-func toString(owner, key string) string {
-	register := flow.NewRegisterID(owner, key)
-	return register.String()
-}
-
 // Get reads a register value from this delta.
 //
 // This function will return nil if the given key has been deleted in this delta.
 // Second return parameters indicated if the value has been set/deleted in this delta
 func (d Delta) Get(owner, key string) (flow.RegisterValue, bool) {
-	value, set := d.Data[toString(owner, key)]
-	return value.Value, set
+	value, set := d.Data[flow.NewRegisterID(owner, key)]
+	return value, set
 }
 
 // Set records an update in this delta.
 func (d Delta) Set(owner, key string, value flow.RegisterValue) {
-	k := toString(owner, key)
-	d.Data[k] = flow.RegisterEntry{
-		Key:   flow.NewRegisterID(owner, key),
-		Value: value,
-	}
+	k := flow.NewRegisterID(owner, key)
+	d.Data[k] = value
 }
 
 // RegisterUpdates returns all registers that were updated by this delta.
 // ids are returned sorted, in ascending order
 func (d Delta) RegisterUpdates() ([]flow.RegisterID, []flow.RegisterValue) {
 	data := make(flow.RegisterEntries, 0, len(d.Data))
-
-	for _, v := range d.Data {
-		data = append(data, v)
+	for k, v := range d.Data {
```

Comment:
I was going to suggest a similar thing, I think we don't need to sort based on both key and values since the `key` is used for the Data map is going to be anyway unique, sorting based on the key but appending the values while we are doing it through the sort method might be the more optimized approach here. I see with the current code you were trying to maintain the same logic but I think in this case is fine to change the sort to be based on the key, considering the `less` implementation of the KeyEntry is only limited to keys.